conda/recipes/librmm/meta.yaml

```diff
@@ -26,13 +26,13 @@ requirements:
     - {{ stdlib("c") }}
   host:
     - cuda-version ={{ cuda_version }}
-    # We require spdlog and fmt (which was de-vendored from spdlog
-    # conda-forge packages in 1.11.0) so that the spdlog headers are not
-    # pulled by CPM and installed as a part of the rmm packages. However,
-    # building against librmm still requires these headers. They are also
-    # added as a run requirement via the packages' run_exports.
+    # We need fmt here for now because the conda spdlog package is hard-coded
```

Comment:
But we removed the spdlog package dependencies…? So this is an “ABI constraint” that is meant to align downstream packages that depend on spdlog, even though fmt is not used/linked as a shared dependency in librmm?

Reply:
Your comment isn't correct, but I'm afraid I don't fully understand your explanation so I may not be able to address your confusion correctly. Basically what is happening is that (because rmm is an interface target) it becomes part of downstream packages build to find spdlog. When that package, say cudf, finds a conda-provided spdlog in the environment, that version of spdlog says to use the fmt shared library even if you use the spdlog_header_only target. Therefore, we actually do have an fmt shared library dependency. That is what I tried to explain in the PR description.

Does that help?

Reply:
Oh I'm sorry I just realized that there is a typo here. The first sentence should read "The conda package for **spdlog** is hard-coded to assume that we use a preexisting fmt library"

Reply:
But the spdlog conda package already has a dependency on fmt. Why not just let that dependency suffice? Why add it to rmm?

Reply:
And spdlog run-exports itself so two different consumers of spdlog will only be usable with compatible spdlog and fmt versions. https://github.com/conda-forge/spdlog-feedstock/blob/97ea289e6733b60e4c619e9cb1e0ebc329358936/recipe/meta.yaml#L18

I’m not convinced we need a fmt dependency here.

Reply:
I realized I never answered this question. The problem is that spdlog could be in the build environment as a transitive dependency even though we no longer have it as a dependency at all, resulting in linkage to fmt without spdlog or fmt being part of the runtime dependency list of our conda package (run exports aren't transitive). But yes, you could also fix this problem by simply putting spdlog back into the dependency list right now.